File: src/ds-visuals/max-heap/application/heap-sort/code/heapsort.c
#include <stdio.h>
#include <conio.h>
#include "heap.c"

void print(int*arr, int n){
	for(int i = 0; i < n; i++)
		printf("%d ",arr[i]);
	printf("\n");
}

void heapSort(int* arr, int n){
	int i;
	Heap heap = createHeap(1, arr[0]);
	for(i = 1; i < n;  i++)
		insert(&heap, arr[i]);
	i = 0;
	while(heap.size > 0){
		arr[i] = delete(&heap);
		i+=1;
	}
}

int main(){
	int arr[] = {4, 13, 6, 2, 87, 21, 65};
	int n = 7;
	print(arr, n);
	heapSort(arr, n);
	print(arr, n);
	return 0;
}

File: src/ds-visuals/max-heap/application/heap-sort/code/heapsort.java
import java.util.*;
//compile max_heap.java in the same directory after rmoving the Main class
class HeapSort{
	int arr[];
	HeapSort(int[] arr){
		this.arr = arr;
	}

	void print(){
		System.out.println(Arrays.toString(this.arr));
	}

	void heapSort(){
		int i;
		Heap heap = new Heap();
		heap.createHeap(this.arr);
		i = 0;
		while(heap.size > 0){
			this.arr[i] = heap.delete();
			i+=1;
		}
	}

	public static void main(String[] args) {
		int arr[] = {4, 13, 6, 2, 87, 21, 65};
		HeapSort heapsort = new HeapSort(arr);
		heapsort.print();
		heapsort.heapSort();
		heapsort.print();
	}
}